src/backend/src/user_profile.rs

```diff
@@ -3,9 +3,24 @@ use ic_cdk::api::time;
 use ic_stable_structures::StableBTreeMap;
 use shared::types::{
     user_profile::{AddUserCredentialError, GetUserProfileError, StoredUserProfile},
-    CredentialType, Version,
+    CredentialType, Timestamp, Version,
 };
 
+fn store_user(
+    principal: StoredPrincipal,
+    timestamp: Timestamp,
+    user: &StoredUserProfile,
+    user_profile_map: &mut StableBTreeMap<(u64, StoredPrincipal), Candid<StoredUserProfile>, VMem>,
+    user_profile_updated_map: &mut StableBTreeMap<StoredPrincipal, u64, VMem>,
+) {
+    if let Some(old_updated) = user_profile_updated_map.get(&principal) {
```

Comment:
Just a note: I still have trouble understanding the purpose of each map when I read the PRs. If we can find better names, I would be in favor of it.

Reply:
I understand that and I'll happily change the names when we have new ones.